### DISCUSSION TITLE
Bad gateway when running data-importer in sub-directory

### DISCUSSION THREAD

**Discussion by kmaasrud** [firefly-iii/firefly-iii#5350] [Firefly III]
I'm running the data-importer in a subdirectory of my Firefly III instance (e.g. Firefly III has URL `firefly.example.com`, and data-importer has url `firefly.example.com/data-importer`). I get the correct landing page, and am prompted to input a client ID. When I do this, I am asked to authorize, but upon doing this I reach a 502... I have tried both inputing the client ID through `.env` and using a personal token as well.

Is this a bug, does the data-importer have to be hosted on its own domain or am I simply missing something obvious? Very thankful if anyone has any tips 😄

**Comment by JC5** [comment]
Running the data import on its own domain yields the best results!

**Reply by kmaasrud** [reply]
I will try that instead! Thanks for answering :smile:

**Reply by kmaasrud** [reply]
I'm now running on a custom domain, and I get the same error. Upon authorizing, I first get a 502 from NGINX, followed by a 500 from Firefly itself. When checking the OAuth clients on Firefly, I get the data-importer listed. I think the issue is with my NGINX reverse proxy configuration, so I understand if this is hard for you to have any solutions to.

Is there anything special I need to have configured for the web server to properly handle the callback? It seems to me like that is where the issue is

**Reply by JC5** [reply]
Not really. There are some nginx examples if you search for them but nothing special is required.

if you get a 500 there should be logs as well. Can you share those?

**Reply by kmaasrud** [reply]
It seems like my proxy buffer size was too small for the HTTP header, so simply increasing it solved the issue. Sorry for bothering you with this, and thanks a lot for your help! :smile:

**Reply by JC5** [reply]
No problem, glad you found the issue!

**Reply by JokerGermany** [reply]
> It seems like my proxy buffer size was too small for the HTTP header, so simply increasing it solved the issue. Sorry for bothering you with this, and thanks a lot for your help! smile

For anyone who is searching for a solution like me:
https://matrix.to/#/!epdtwMcKTscMlFxeBi:gitter.im/$fytGAXaexbmXi4lfmp8f2fDDcwqOM3gp-C4boLin0rc?via=gitter.im&via=matrix.org&via=beckmeyer.us
> \# Tune Nginx buffers # 
> server {
> proxy_busy_buffers_size   512k;
> proxy_buffers   4 512k;
> proxy_buffer_size   256k;
> \# rest of the nginx config below #
> }

(Needs to be inserted for the importer)

**Reply by mgundelfinger** [reply]
> proxy_busy_buffers_size 512k;
proxy_buffers 4 512k;
proxy_buffer_size 256k;

when using nginx proxy manager, just adding these three lines in the advanced section fixed it for me! thanks a lot!
